app/src/main/java/com/example/fitjournal/MainActivity.kt

```diff
@@ -60,9 +68,8 @@ class MainActivity : ComponentActivity() {
             FitJournalTheme {
                 val navController = rememberNavController()
                 val snackBarState = remember { SnackbarHostState() }
-                var showChildFabs by remember {
-                    mutableStateOf(false)
-                }
+                var showChildFabs by remember { mutableStateOf(false) }
+                var selectedWorkoutDetail: WorkoutDetail? by remember { mutableStateOf(null) }
```

Comment:
What is the point of this variable?

Reply:
I need this so I know what workout you selected from the list of workouts that way when we navigate to the view where you will enter your sets, weight and repetitions I can pass the selected workout and display the correct textfields and data.

`           mainScreen = { mainModifier ->
                                    JournalEntryScreen(
                                        modifier = mainModifier,
                                        selectedJournalEntry = { selectedWorkoutDetail = it },
                                        navigateToDestination = { navigation ->
                                            navigateToDestination(
                                                navigationInterface = navigation,
                                                navController = navController
                                            )
                                        }
                                    )
   `
   
   See here
   
   `
                        composable(Route.JOURNAL_ENTRY_DETAILS) {
                            val viewModel: JournalEntryDetailsViewModel = viewModel(factory = JournalEntryDetailsViewModelFactory(selectedWorkoutDetail))
                            LaunchedEffect(Unit) {
                                bottomBarVisibility.value = false
                            }
                            AppScreen(
                                showMainFabIcon = false,
                                modifier = Modifier,
                                snackBarHostState = snackBarState,
                                mainScreen = { mainModifier ->
                                    JournalEntryDetailsScreen(
                                        modifier = mainModifier,
                                        viewModel = viewModel
                                    )
                                },
`

Reply:
If there is a better way to handle this navigation or a different way should be created for this internal navigation then we can talk and I can implement that.

Reply:
natural compose navigation should have an ability to pass in arguments when creating navigation routes.. may need to look into it

Reply:
myNewList.foreach{ childList ->
if (childList.isNotEmpty()){
     sticky header (workoutType)
       WorkoutList
   }
}